### PR TITLE
DX-2240 Updated Sample and Gemfile

### PR DESCRIPTION
**Pull request by ckoegel**

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    messaging_basic_auth_user_name: BW_USERNAME,
+    messaging_basic_auth_password: BW_PASSWORD
+    # environment: Environment::CUSTOM, #Optional - Used for custom base URLs
+    # base_url: "https://d6979da481772c167be0edcd10eb64d7.m.pipedream.net" #Optional - Custom base URL set here
+)
+messaging_client = bandwidth_client.messaging_client.client
+
+account_id = BW_ACCOUNT_ID
+
+post '/callbacks/outbound/messaging' do
```
Comment (abaldwin-Bandwidth) [inline app.rb:26]:
```suggestion
post '/example/send/message' do
```

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    #Make a post request to this url to send outbound MMS with media
```
Comment (abaldwin-Bandwidth) [inline app.rb]:
```suggestion
    #Create and send a message when someone POSTs to this endpoint
```

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    body.to = [data["to"]]
+    body.from = BW_NUMBER
+    body.text = data["text"]
+    body.media = ["https://cdn2.thecatapi.com/images/MTY3ODIyMQ.jpg"]
+
+    messaging_client.create_message(account_id, body)
+    return ''
+end
+
+post '/callbacks/outbound/messaging/status' do
```
Comment (abaldwin-Bandwidth) [inline app.rb:40]:
```suggestion
post '/callbacks/outbound/messaging/status' do
#Create a callback endpoint to receive status updates on sent messages
```

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        when "message-failed"
+            puts "For MMS and Group Messages, you will only receive this callback if you have enabled delivery receipts on MMS."
+        else
+            puts "Message type does not match endpoint. This endpoint is used for message status callbacks only."
+        end
+    return ''
+end
+
+post '/callbacks/inbound/messaging' do
+    #This URL handles inbound messages.
```
Comment (abaldwin-Bandwidth) [inline app.rb]:
```suggestion
    #Create a callback endpoint to receive inbound messages
```

**Review by ajrice6713 — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -1,2 +1,69 @@
...
+BW_NUMBER                            # The Bandwidth phone number involved with this application
+LOCAL_PORT                           # The port number you wish to run the sample on
+```
+
+# Callback URLs
+
+For a detailed introduction to Bandwidth Callbacks see https://dev.bandwidth.com/guides/callbacks/callbacks.html
+
+Below are the callback paths:
+* `/callbacks/outbound/messaging` For Sending Media Messages
```
Comment (abaldwin-Bandwidth) [inline README.md:57]:
```suggestion

```
This isn't an actual callback that a client would need to implement, it's just an example endpoint we're creating, so we shouldn't talk about it like it's some callback URL for our API

File: app.rb [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    #If the inbound message contains media, that media is downloaded
+    data = JSON.parse(request.body.read)
+    if data[0]["type"] == "message-received"
+        puts "Message received"
+        puts "To: " + data[0]["message"]["to"][0] + "\nFrom: " + data[0]["message"]["from"] + "\nText: " + data[0]["message"]["text"]
+        if data[0]["message"].key?("media")
+            data[0]["message"]["media"].each do |media|
+                media_id = media.split("/").last(3)
```
Comment (abaldwin-Bandwidth) [inline app.rb:65]: Maybe offer up some explanation about why the media_id is being manipulated

File: README.md (hunk 5 above)
Comment (ckoegel) [reply README.md:57]: i agree, but this is the same endpoint used for all of our other sample apps, and they all document it this way, so we'd probably have to change them all to be consistent

File: app.rb (hunk 1 above)
Comment (hamermike) [reply app.rb:26]: I'd leave this as it is since this is how it is for all the samples that are defined.

**Review by ajrice6713 — APPROVED** [review]

**Review by hamermike — APPROVED** [review]

**Pull request merged**